src/Adapter/Product/Image/Repository/ProductImageMultiShopRepository.php

```diff
@@ -506,4 +541,37 @@ public function getImageById(ImageId $imageId): Image
 
         return $image;
     }
+
+    /**
+     * @return ImageType[]
+     */
+    public function getProductImageTypes(): array
+    {
+        try {
+            $results = ImageType::getImagesTypes('products');
+        } catch (PrestaShopException $e) {
+            throw new CoreException('Error occurred when trying to get product image types');
+        }
+
+        if (!$results) {
+            return [];
+        }
+
+        $imageTypes = [];
+        foreach ($results as $result) {
+            $imageType = new ImageType();
+            $imageType->id = (int) $result['id_image_type'];
+            $imageType->name = $result['name'];
+            $imageType->width = (int) $result['width'];
+            $imageType->height = (int) $result['height'];
+            $imageType->products = (bool) $result['products'];
+            $imageType->categories = (bool) $result['categories'];
+            $imageType->manufacturers = (bool) $result['manufacturers'];
+            $imageType->suppliers = (bool) $result['suppliers'];
+            $imageType->stores = (bool) $result['stores'];
+            $imageTypes[] = $imageType;
```

Comment:
```suggestion
            $imageType = new ImageType();
            $imageType->hydrate($result);
            $imageTypes[] = $imageType;
```

maybe you could do this instead, although I'm not sure why you need to return array of objects

Reply:
Actually, I just copied this from the original `ProductImageRepository` service, I didn't think about what I copied since the purpose of this PR is not to refacto this part, although I think returning object makes sense I'm not sure it should be some ObjectModel, and the way they are created can definitely be improved but again not the point here

You're not the first one to point at stuff about this method and the fact that I turned this service as public, if it really bothers you all I can just revert this change as it's not mandatory for this PR and I don't want it to be blocked for parallel issues

Reply:
Sure, we have a refacto that aims at merging `ProductImageRepository` with `ProductImageMultiShopRepository` so I'll try and improve it or at least add the todo at that moment

Reply:
noted.
Related pr https://github.com/PrestaShop/PrestaShop/pull/31611